core/src/solution_submission/retry.rs

```diff
@@ -99,9 +106,44 @@ fn gas_price(estimated_price: U256, price_increase_count: u32, cap: U256) -> U25
     cap.min(U256::from(new_price as u128))
 }
 
+/// The output type of futures used in `select_all`.
+enum SelectFutureOutput {
```

Comment:
I called it output because in the future trait the type is also called output. I'm fine with your suggestion too though.

Reply:
To me `Output` is any future output, however in this case its more of a `Result` in that it can be successful or some sort of failure. Actually, I would slightly prefer to use `Result<(), SolutionDeadlineError>` with:
```rust
enum SolutionDeadlineError {
    SolutionSubmission(MethodError),
    Timeout,
}
```

This is mostly for the goodies with the `Result` type.

Reply:
The original point of this enum is that `select_all` requires all the futures to have the same output type. So this is really just enum over the two possible types and not a result. I don't think we would be using any result goodies and it would be more code to inspect the error type when doing it that way.